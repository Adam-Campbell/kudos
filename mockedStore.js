export const store = {
    articles: {
        article_A: {
            _id: 'article_A',
            title: 'The title for Article A',
            description: 'The description for Article A.',
            image: 'https://example.com/article_a.jpg',
            text: 'The main body of Article A.',
            kudos: 7,
            author: 'user_A',
            category: 'category_A',
            commentIds: ['comment_A', 'comment_C']
        },
        article_B: {
            _id: 'article_B',
            title: 'The title for Article B',
            description: 'The description for Article B.',
            image: 'https://example.com/article_b.jpg',
            text: 'The main body of Article B.',
            kudos: 2,
            author: 'user_B',
            category: 'category_B',
            commentIds: ['comment_B', 'comment_D']
        },
        article_C: {
            _id: 'article_C',
            title: 'The title for Article C',
            description: 'The description for Article C.',
            image: 'https://example.com/article_c.jpg',
            text: 'The main body of Article C.',
            kudos: 0,
            author: 'user_A',
            category: 'category_B',
            commentIds: []
        }
    },
    users: {
        user_A: {
            _id: 'user_A',
            followers: 1,
            following: 0,
            username: 'Joe Bloggs',
            avatar: 'https://example.com/user_a.jpg',
            bio: 'The bio for User A.',
            postIds: ['article_A', 'article_C'],
            commentIds: ['comment_A', 'comment_B'],
            kudosIds: ['article_A'],
            highlightIds: ['article_A']
        },
        user_B: {
            _id: 'user_B',
            followers: 0,
            following: 1,
            username: 'Jane Smith',
            avatar: 'https://example.com/user_b.jpg',
            bio: 'The bio for User B.',
            postIds: ['article_B'],
            commentIds: ['comment_C'],
            kudosIds: ['article_B']
        }
    },
    comments: {
        comment_A: {
            _id: 'comment_A',
            text: 'The text for Comment A.',
            author: 'user_A',
            discussion: 'article_A',
            createdAt: 'Wed, 06 Jun 2018 08:07:53 GMT',
            parents: ['comment_A']
        },
        comment_B: {
            _id: 'comment_B',
            text: 'The text for Comment B.',
            author: 'user_A',
            discussion: 'article_B',
            createdAt: 'Wed, 06 Jun 2018 10:32:06 GMT',
            parents: ['comment_B']
        },
        comment_C: {
            _id: 'comment_C',
            text: 'The text for Comment C.',
            author: 'user_B',
            discussion: 'article_A',
            createdAt: 'Wed, 06 Jun 2018 12:21:27 GMT',
            parents: ['comment_A', 'comment_B']
        },
        comment_D: {
            _id: 'comment_D',
            text: 'Comment deleted by user.',
            discussion: 'article_B',
            createdAt: 'Wed, 06 Jun 2018 13:46:12 GMT',
            author: null,
            discussion: 'article_B',
            parents: ['comment_D']
        }
    },
    highlights: {
        highlight_A: {
            _id: 'highlight_A',
            post: 'article_A',
            user: 'user_A',
            excerpt: 'An excerpt from Article A.'
        }
    },
    currentUser: {
        isLoggedIn: true,
        token: 'abc123',
        _id: 'user_A',
        follows: ['user_B'],
        email: 'example@example.com'
    },
    categories: {
        all: ['article_A', 'article_B', 'article_C'],
        category_A: ['article_A'],
        category_B: ['article_B', 'article_C']
    },
    errors: {
        signUpDuplicateError: false,
        emailNotFound: false
    },
    successes: {
        passwordResetEmailSent: false
    }
}