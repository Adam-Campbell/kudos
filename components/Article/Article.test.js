import { shallow } from 'enzyme';
import { ArticleContainer } from './index';
import { store } from '../../mockedStore';
const article_id = 'article_A';
const altArticle_id = 'article_B';

describe('<ArticleContainer/>', () => {
    test('renders correctly if user is author of article', () => {
        const container = shallow(
            <ArticleContainer 
                article_id={article_id}
                articles={store.articles}
                isLoggedIn={store.currentUser.isLoggedIn}
                currentUser_id={store.currentUser._id}
            />
        );
        // create wrapper for the presentational component returned
        const presenter = container.first().shallow();
        // check snapshots for container and presentational components
        expect(container).toMatchSnapshot();
        expect(presenter).toMatchSnapshot();
        // check correct props are passed to the presentational component
        expect(container.find('Article').prop('isAuthor')).toBe(true);
    });
    test("renders correctly if user isn't author of article", () => {
        const container = shallow(
            <ArticleContainer 
                article_id={altArticle_id}
                articles={store.articles}
                isLoggedIn={store.currentUser.isLoggedIn}
                currentUser_id={store.currentUser._id}
            />
        );
        // create wrapper for the presentational component returned
        const presenter = container.first().shallow();
        // check snapshots for container and presentational components
        expect(container).toMatchSnapshot();
        expect(presenter).toMatchSnapshot();
        // check correct props are passed to the presentational component
        expect(container.find('Article').prop('isAuthor')).toBe(false);
    });
});
