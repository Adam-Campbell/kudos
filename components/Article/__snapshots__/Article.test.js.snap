// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleContainer/> renders correctly if user is author of article 1`] = `
<Article
  articleDescription="The description for Article A."
  articleImage="https://example.com/article_a.jpg"
  articleKudos={7}
  articleText="The main body of Article A."
  articleTitle="The title for Article A"
  article_id="article_A"
  commentIds={
    Array [
      "comment_A",
      "comment_C",
    ]
  }
  isAuthor={true}
  isLoggedIn={true}
/>
`;

exports[`<ArticleContainer/> renders correctly if user is author of article 2`] = `
<ArticleOuter>
  <ArticleIntroContainer
    tight={true}
  >
    <Link
      as="/edit-post/article_A"
      href="/edit-post?post=article_A"
      passHref={true}
    >
      <EditArticleButton>
        Edit this article
      </EditArticleButton>
    </Link>
    <Connect(AuthorBlockContainer)
      article_id="article_A"
    />
    <ArticleHeader>
      <ArticleTitle>
        The title for Article A
      </ArticleTitle>
      <ArticleDescription>
        The description for Article A.
      </ArticleDescription>
      <KudosStatsContainer>
        <KudosStat>
          <span>
            7
          </span>
           Kudos
        </KudosStat>
        <Connect(KudosButtonContainer)
          article_id="article_A"
        />
      </KudosStatsContainer>
    </ArticleHeader>
  </ArticleIntroContainer>
  <section>
    <Wrapper
      wide={true}
    >
      <ArticleFigure>
        <ArticleImage
          imageUrl="https://example.com/article_a.jpg"
        />
        <ArticleImageCaption>
          I haven't added these yet...
        </ArticleImageCaption>
      </ArticleFigure>
    </Wrapper>
    <Wrapper
      tight={true}
    >
      <ArticleParagraph>
        The main body of Article A.
      </ArticleParagraph>
      <ArticleQuote>
        <ArticleQuoteText>
          “This is where my inspirational quotes would go IF I HAD ANY”
        </ArticleQuoteText>
      </ArticleQuote>
    </Wrapper>
  </section>
  <section>
    <Wrapper
      tight={true}
    >
      <Connect(ArticleCommentBoxContainer)
        article_id="article_A"
      />
      <Connect(CommentSelector)
        comment_id="comment_A"
        key="comment_A"
      />
      <Connect(CommentSelector)
        comment_id="comment_C"
        key="comment_C"
      />
    </Wrapper>
  </section>
</ArticleOuter>
`;

exports[`<ArticleContainer/> renders correctly if user isn't author of article 1`] = `
<Article
  articleDescription="The description for Article B."
  articleImage="https://example.com/article_b.jpg"
  articleKudos={2}
  articleText="The main body of Article B."
  articleTitle="The title for Article B"
  article_id="article_B"
  commentIds={
    Array [
      "comment_B",
      "comment_D",
    ]
  }
  isAuthor={false}
  isLoggedIn={true}
/>
`;

exports[`<ArticleContainer/> renders correctly if user isn't author of article 2`] = `
<ArticleOuter>
  <ArticleIntroContainer
    tight={true}
  >
    <Connect(AuthorBlockContainer)
      article_id="article_B"
    />
    <ArticleHeader>
      <ArticleTitle>
        The title for Article B
      </ArticleTitle>
      <ArticleDescription>
        The description for Article B.
      </ArticleDescription>
      <KudosStatsContainer>
        <KudosStat>
          <span>
            2
          </span>
           Kudos
        </KudosStat>
        <Connect(KudosButtonContainer)
          article_id="article_B"
        />
      </KudosStatsContainer>
    </ArticleHeader>
  </ArticleIntroContainer>
  <section>
    <Wrapper
      wide={true}
    >
      <ArticleFigure>
        <ArticleImage
          imageUrl="https://example.com/article_b.jpg"
        />
        <ArticleImageCaption>
          I haven't added these yet...
        </ArticleImageCaption>
      </ArticleFigure>
    </Wrapper>
    <Wrapper
      tight={true}
    >
      <ArticleParagraph>
        The main body of Article B.
      </ArticleParagraph>
      <ArticleQuote>
        <ArticleQuoteText>
          “This is where my inspirational quotes would go IF I HAD ANY”
        </ArticleQuoteText>
      </ArticleQuote>
    </Wrapper>
  </section>
  <section>
    <Wrapper
      tight={true}
    >
      <Connect(ArticleCommentBoxContainer)
        article_id="article_B"
      />
      <Connect(CommentSelector)
        comment_id="comment_B"
        key="comment_B"
      />
      <Connect(CommentSelector)
        comment_id="comment_D"
        key="comment_D"
      />
    </Wrapper>
  </section>
</ArticleOuter>
`;
